src/library/serato/seratofeature.cpp

```diff
@@ -0,0 +1,1065 @@
+// seratofeature.cpp
+// Created 2020-01-31 by Jan Holthuis
+
+#include "library/serato/seratofeature.h"
+
+#include <QMap>
+#include <QMessageBox>
+#include <QSettings>
+#include <QStandardPaths>
+#include <QtDebug>
+
+#include "engine/engine.h"
+#include "library/library.h"
+#include "library/librarytablemodel.h"
+#include "library/missingtablemodel.h"
+#include "library/queryutil.h"
+#include "library/trackcollection.h"
+#include "library/trackcollectionmanager.h"
+#include "library/treeitem.h"
+#include "library/dao/trackschema.h"
+#include "track/beatfactory.h"
+#include "track/cue.h"
+#include "track/keyfactory.h"
+#include "util/assert.h"
+#include "util/color/color.h"
+#include "util/db/dbconnectionpooled.h"
+#include "util/db/dbconnectionpooler.h"
+#include "util/file.h"
+#include "util/sandbox.h"
+#include "waveform/waveform.h"
+#include "widget/wlibrary.h"
+#include "widget/wlibrarytextbrowser.h"
+
+namespace {
+
+// Serato Database Field IDs
+// The "magic" value is the short 4 byte ascii code intepreted as quint32, so
+// that we can use the value in a switch statement instead of going through
+// a strcmp if/else ladder.
+enum class FieldId : quint32 {
+    Version = 0x7672736e,        // vrsn
+    Track = 0x6f74726b,          // otrk
+    FileType = 0x74747970,       // ttyp
+    FilePath = 0x7066696c,       // pfil
+    SongTitle = 0x74736e67,      // tsng
+    Artist = 0x74617274,         // tart
+    Album = 0x74616c62,          // talb
+    Genre = 0x7467656e,          // tgen
+    Comment = 0x74636f6d,        // tcom
+    Grouping = 0x74677270,       // tgrp
+    Label = 0x746c626c,          // tlbl
+    Year = 0x74747972,           // ttyr
+    Length = 0x746c656e,         // tlen
+    Bitrate = 0x74626974,        // tbit
+    SampleRate = 0x74736d70,     // tsmp
+    Bpm = 0x7462706d,            // tbpm
+    DateAddedText = 0x74616464,  // tadd
+    DateAdded = 0x75616464,      // uadd
+    Key = 0x746b6579,            // tkey
+    BeatgridLocked = 0x6262676c, // bbgl
+    FileTime = 0x75746d65,       // utme
+    Missing = 0x626d6973,        // bmis
+    Sorting = 0x7472736f,        // osrt
+    ReverseOrder = 0x62726576,   // brev
+    ColumnTitle = 0x6f766374,    // ovct
+    ColumnName = 0x7476636e,     // tvcn
+    ColumnWidth = 0x74766377,    // tvcw
+    TrackPath = 0x7074726b,      // ptrk
```

Comment:
I think you use Location elsewhere. How about use here TrackLocation?

Reply:
The field is `ptrk` and the first char `p` denotes a path. I tried to keep the enum member names close to the original serato naming.